bokeh/plotting/helpers.py

```diff
@@ -505,38 +549,16 @@ def func(self, *args, **kwargs):
             _update_legend(self, legend_item_label, glyph_renderer)
 
         for tool in self.select(type=BoxSelectTool):
-            # this awkward syntax is needed to go through Property.__set__ and
-            # therefore trigger a change event. With improvements to Property
-            # we might be able to use a more natural append() or +=
-            tool.renderers = tool.renderers + [glyph_renderer]
+            tool.renderers.append(glyph_renderer)
+
+        self.renderers.append(glyph_renderer)
 
-        # awkward syntax for same reason mentioned above
-        self.renderers = self.renderers + [glyph_renderer]
```

Comment:
The property containers now detect changes like this correctly, so these can be simplified